extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -393,47 +400,51 @@ export class Visibility {
       this.resources_.push(resource);
     });
 
-    // Hidden trigger
-    if (!shouldBeVisible && !this.visibilityListenerRegistered_) {
+    if (!this.visibilityListenerRegistered_) {
       this.viewer_.onVisibilityChanged(() => {
-        if (!this.viewer_.isVisible()) {
-          this.onDocumentHidden_();
-        }
+        this.onDocumentVisibilityChange_(this.viewer_.isVisible());
       });
       this.visibilityListenerRegistered_ = true;
     }
   }
 
-  /** @private */
-  onIntersectionChange_(change) {
-    const resource =
-        this.resourcesService_.getResourceForElement(change.target);
+  /**
+   * @param {!Element} target
+   * @param {number} visible
```

Comment:
I was following the naming in V1. it makes sense to be consistent for now. we can rename after the removal of V1 code.